fhir-search/src/main/java/com/ibm/fhir/search/util/SearchUtil.java

```diff
@@ -1589,6 +1605,168 @@ private static QueryParameter parseChainedParameter(Class<?> resourceType, Strin
         return rootParameter;
     }
 
+    /**
+     * Transforms the passed string representing reverse chain search criteria, into
+     * an actual chain of QueryParameter objects. This method consumes strings of this form:
+     * <pre>
+     *      +-------------------------------------------------------------------+
+     *      |                                                                   |
+     *      V                                                                   |
+     * >>---+--- "_has:{referenced-by-resource-type}:{reference-parameter}:" ---+--- "{search-parameter}" ---><
+     * </pre>
+     * See the FHIR specification for details:
+     * <a href="https://www.hl7.org/fhir/search.html#has</a>
+     *
+     * @param resourceType
+     *          Search type.
+     * @param reverseChainParameterString
+     *          Reverse chain search parameter string.
+     * @param valuesString
+     *          String containing the final search value.
+     * @return QueryParameter
+     *          The root of a parameter chain for the reverse chain criteria.
+     */
+    private static QueryParameter parseReverseChainedParameter(Class<?> resourceType, String reverseChainParameterString, String valuesString) throws Exception {
+
+        QueryParameter rootParameter = null;
+
+        try {
+            // Strip leading '_has:' and then split by ':_has:'
+            List<String> components = Arrays.asList(reverseChainParameterString
+                .replaceFirst(SearchConstants.HAS + SearchConstants.COLON_DELIMITER_STR, "")
+                .split(SearchConstants.COLON_DELIMITER_STR + SearchConstants.HAS + SearchConstants.COLON_DELIMITER_STR));
```

Comment:
I'm almost positive in modern compilers this ends up getting inlined, however, it's worth making your own static final constant.

Reply:
created constant